s3/s3_test.go

```diff
@@ -0,0 +1,95 @@
+package s3
+
+import (
+	"flag"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/spf13/afero/test"
+)
+
+var s3bucket string
+
+func init() {
```

Comment:
This should be done as an integration test. Instead, you could Money Patch s3 methods (mock cannot really be done).

```go
// my_test.go
...
monkey.PatchInstanceMethod(reflect.TypeOf(s3Session), "GetObject", s3_mocks.GetObjectMockFn())
...
```

```go
// mocks_s3.go

type ClosingBuffer struct {
    *bytes.Buffer
}

func (cb *ClosingBuffer) Close() error {
	return nil
}

func GetObjectMockFn() func(_ *s3.S3, _ *s3.GetObjectInput) (*s3.GetObjectOutput, error) {
	return func(_ *s3.S3, _ *s3.GetObjectInput) (*s3.GetObjectOutput, error) {
		cb := &ClosingBuffer{bytes.NewBufferString(`This is the content of a file`)}

		var rc io.ReadCloser
		rc = cb
		rc.Close()

		return &s3.GetObjectOutput{
			Body: rc,
		}, nil
	}
}